src/logcounter.hpp

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright (c) 2021 Staz Modrzynski
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+#ifndef WEBRTC_SERVER_LOGCOUNTER_HPP
+#define WEBRTC_SERVER_LOGCOUNTER_HPP
+
+#include "threadpool.hpp"
+#include "include.hpp"
+
+namespace rtc {
+class LogCounter {
+private:
+    plog::Severity mSeverity;
+    std::string mText;
+    std::chrono::steady_clock::duration mDuration;
+
+    std::atomic<int> mCount = 0;
+
+    std::shared_ptr<std::mutex> mIsValidMutex;
+    std::shared_ptr<bool> mIsValid;
```

Comment:
I think this works, however you don't actually need the mutex since `mCount` is atomic.

Just a suggestion: you could simplify by putting `mSeverity`, `mText`, `mDuration`, and `mCount` directly in a structure held by a `shared_ptr`. Then you only need to pass a `weak_ptr` to the task, the validity will be given by whether the `weak_ptr` can be locked or not.

Reply:
:+1: